insonmnia/logging/trace.go

```diff
@@ -0,0 +1,32 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opentracing/basictracer-go"
+	"github.com/opentracing/opentracing-go"
+	"go.uber.org/zap"
+)
+
+const (
+	traceIdFieldKey = "span.trace_id"
+	spanIdFieldKey  = "span.span_id"
+)
+
+func WithTrace(ctx context.Context, log *zap.Logger) *zap.Logger {
+	span := opentracing.SpanFromContext(ctx)
+	if span == nil {
+		return log
+	}
+
+	spanContext, ok := span.Context().(basictracer.SpanContext)
+	if ok && spanContext.Sampled {
```

Comment:
I doubt that check for Sampled is required here. Did you check if all records in this manner do have corresponding tracing fields?